parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh

```diff
@@ -107,19 +107,37 @@ installStandaloneContainerd() {
     if semverCompare ${CURRENT_VERSION:-"0.0.0"} ${CONTAINERD_VERSION}; then
         echo "currently installed containerd version ${CURRENT_VERSION} is greater than (or equal to) target base version ${CONTAINERD_VERSION}. skipping installStandaloneContainerd."
     else
-        echo "installing containerd version ${CONTAINERD_VERSION}"
-        removeMoby
-        removeContainerd
-        updateAptWithMicrosoftPkg
-        apt_get_install 20 30 120 moby-containerd=${CONTAINERD_VERSION}* --allow-downgrades || exit $ERR_CONTAINERD_INSTALL_TIMEOUT
+        CONTAINERD_DEB_TMP="moby-containerd_${CONTAINERD_VERSION/-/\~}+azure-1_amd64.deb"
+        CONTAINERD_DEB_FILE="$CONTAINERD_DOWNLOADS_DIR/${CONTAINERD_DEB_TMP}"
+        # first we check if the vhd has the target version saved locally - exercised for containerd override scenarios during node provisiong
+        if [[ -f ${CONTAINERD_DEB_FILE} ]]; then
+            echo "installing containerd version ${CONTAINERD_VERSION} from cached file ${CONTAINERD_DEB_FILE}"
+            installStandaloneContainerdFromFile ${CONTAINERD_DEB_FILE} || exit $ERR_CONTAINERD_INSTALL_TIMEOUT
```

Comment:
we also need the below two , right?
 removeMoby
            removeContainerd

Reply:
yes, moving up